cura/UI/WelcomePagesModel.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) 2019 Ultimaker B.V.
+# Cura is released under the terms of the LGPLv3 or higher.
+import os
+from typing import TYPE_CHECKING, Optional
+
+from PyQt5.QtCore import QUrl, Qt, pyqtSlot
+
+from UM.Qt.ListModel import ListModel
+from UM.Resources import Resources
+
+if TYPE_CHECKING:
+    from PyQt5.QtCore import QObject
+
+
+class WelcomePagesModel(ListModel):
+
+    IdRole = Qt.UserRole + 1  # Page ID
+    PageUrlRole = Qt.UserRole + 2  # URL to the page's QML file
+    NextPageIdRole = Qt.UserRole + 3  # The next page ID it should go to
```

Comment:
We could consider adding some sort of "weights" to this so that plug-ins can define their page to be in a specific position in the wizard without conflicting each other. Maybe that is out of scope, though, and a different ticket can be made. You decide.

Reply:
At first I was considering this too, but I decided to go simple because we don't know yet what plugins would want to add. I personally would leave it before we have a better idea on this.